weave

```diff
@@ -746,6 +746,53 @@ get_ipam_cidr() {
     fi
 }
 
+######################################################################
+# weave proxy helpers
+######################################################################
+
+
+# TODO: Handle relative paths for args
+# TODO: Handle args with spaces
+proxy_args() {
+    PROXY_VOLUMES=""
+    PROXY_ARGS=""
+    while [ $# -gt 0 ]; do
+        case "$1" in
+          --tlscacert)
+            shift
+            PROXY_VOLUMES="$PROXY_VOLUMES -v $1:/home/weave/tls/ca.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlscacert /home/weave/tls/ca.pem"
+            ;;
+          --tlscacert=*)
+            PROXY_VOLUMES="$PROXY_VOLUMES -v ${1#*=}:/home/weave/tls/ca.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlscacert /home/weave/tls/ca.pem"
+            ;;
+          --tlscert)
+            shift
+            PROXY_VOLUMES="$PROXY_VOLUMES -v $1:/home/weave/tls/cert.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlscert /home/weave/tls/cert.pem"
+            ;;
+          --tlscert=*)
+            PROXY_VOLUMES="$PROXY_VOLUMES -v ${1#*=}:/home/weave/tls/cert.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlscert /home/weave/tls/cert.pem"
+            ;;
+          --tlskey)
+            shift
+            PROXY_VOLUMES="$PROXY_VOLUMES -v $1:/home/weave/tls/key.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlskey /home/weave/tls/key.pem"
+            ;;
+          --tlskey=*)
+            PROXY_VOLUMES="$PROXY_VOLUMES -v ${1#*=}:/home/weave/tls/key.pem"
+            PROXY_ARGS="$PROXY_ARGS --tlskey /home/weave/tls/key.pem"
+            ;;
+          *)
+            PROXY_ARGS="$PROXY_ARGS $1"
+            ;;
+        esac
+        shift
+    done
+}
+
```

Comment:
A `tls_arg` function would do wonders here. You've invoke it s.t. like this:
````
        --tlscacert)
            shift
            tls_arg "cacert" "ca" $1
            ;;
          --tlscacert=*)
            tls_arg "cacert" "ca" ${1#*=}
            ;;
````

And you wouldn't even need the 2nd param if the CA cert file was named `cacert.pem` instead of `ca.pem`.